### PR TITLE
expression: tiny refactor around the vectorized testing framework

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve? 
1. Introduce `defaultGener`.
2. Check warnings generated by expressions.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/bench_test.go [hunk 1]
```diff
@@ -200,6 +200,52 @@ type dataGenerator interface {
 	gen() interface{}
 }
 
+type defaultGener struct {
+	nullRation float64
+	eType      types.EvalType
+}
+
+func (g *defaultGener) gen() interface{} {
```
Comment (XuHuaiyu) [inline expression/bench_test.go]: Do hybrid types needs to be handled here?

File: expression/bench_test.go [hunk 2]
```diff
@@ -611,6 +608,14 @@ func (s *testEvaluatorSuite) TestVectorizedBuiltinFunc(c *C) {
 			default:
 				c.Fatal(fmt.Sprintf("evalType=%v is not supported", testCase.retEvalType))
 			}
+
+			// check warnings
+			totalWarns := ctx.GetSessionVars().StmtCtx.WarningCount()
+			c.Assert(2*vecWarnCnt, Equals, totalWarns)
+			warns := ctx.GetSessionVars().StmtCtx.GetWarnings()
+			for i := 0; i < int(vecWarnCnt); i++ {
+				c.Assert(warns[i].Err.Error(), Equals, warns[i+int(vecWarnCnt)].Err.Error())
```
Comment (XuHuaiyu) [inline expression/bench_test.go]: Will `Error()` return the error stack message?
If so, will it be better to use `c.Assert(ErrorEqual(err1, err2), IsTrue)` here?

File: expression/bench_test.go (hunk 1 above)
Comment (qw4990) [reply expression/bench_test.go]: How about supporting hybrid types later or letting our contributors to support this feature?

File: expression/bench_test.go (hunk 2 above)
Comment (qw4990) [reply expression/bench_test.go]: You're right.
It is fixed.

File: expression/bench_test.go [hunk 3]
```diff
@@ -17,6 +17,7 @@ package expression
 
 import (
 	"fmt"
+	"github.com/pingcap/parser/terror"
```
Comment (XuHuaiyu) [inline expression/bench_test.go]: move this down

**Pull request merged**